apps/gitness/src/pages/project-settings/project-settings-members-page.tsx

```diff
@@ -0,0 +1,181 @@
+import React, { useState } from 'react'
+import { Spacer, Text, ListActions, SearchBox, Button } from '@harnessio/canary'
+
+import {
+  SandboxLayout,
+  SkeletonList,
+  NoData,
+  MembersList,
+  FormEditMemberDialog,
+  FormDeleteMemberDialog,
+  useCommonFilter,
+  PaginationComponent
+} from '@harnessio/playground'
+import {
+  useMembershipListQuery,
+  TypesMembershipUser,
+  EnumMembershipRole,
+  MembershipListQueryQueryParams
+} from '@harnessio/code-service-client'
+import { useGetSpaceURLParam } from '../../framework/hooks/useGetSpaceParam'
+import { Link } from 'react-router-dom'
+import { timeAgoFromEpochTime } from '../pipeline-edit/utils/time-utils'
+import { useQueryState, parseAsInteger } from 'nuqs'
+import { PageResponseHeader } from '../../types'
+
+const filterOptions = [{ name: 'Filter option 1' }, { name: 'Filter option 2' }, { name: 'Filter option 3' }]
+const sortOptions = [{ name: 'Sort option 1' }, { name: 'Sort option 2' }, { name: 'Sort option 3' }]
+
+const ProjectSettingsMemebersPage = () => {
+  const space_ref = useGetSpaceURLParam()
+  const [totalMembers, setTotalMembers] = useState<number | null>(null)
+  const [isSubmitting, setIsSubmitting] = useState(false)
+  const [isDeleting, setIsDeleting] = useState(false)
+  const [dialogState, setDialogState] = useState({
+    isDialogEditOpen: false,
+    isDialogDeleteOpen: false,
+    selectedMember: null as { display_name: string; role: string; email: string } | null
+  })
+
+  const { sort, query: currentQuery } = useCommonFilter<MembershipListQueryQueryParams['sort']>()
+  const [query, _] = useQueryState('query', { defaultValue: currentQuery || '' })
+  const [page, setPage] = useQueryState('page', parseAsInteger.withDefault(1))
+
+  const { isLoading, data: { body: members, headers } = {} } = useMembershipListQuery(
+    { space_ref: space_ref ?? '', queryParams: { query, sort } },
+    {
+      onSuccess: ({ body: members }) => {
+        setTotalMembers(members.length) // Update total members count
+        console.log(members.length)
+      }
+    }
+  )
+
+  const totalPages = parseInt(headers?.get(PageResponseHeader.xTotalPages) || '')
+
+  const handleDelete = () => {
+    setIsDeleting(true)
+    setTimeout(() => {
+      setIsDeleting(false)
+      setDialogState(prev => ({ ...prev, isDialogDeleteOpen: false, selectedMember: null }))
+    }, 2000)
+  }
```

Comment:
This is just a mocked delete action right?

Reply:
yes